README.md
Vagrant-Setup [ Luis Felipe && EspecializaTI ]
===========

Servidor Web (Linux, Apache, PostgreSql, PHP)

Configuração do Vagrant (com provisionamento em Shell Script) para criar uma máquina virtual (Ubuntu Server 16.04 64 Bits) de desenvolvimento em PHP.

### Pacotes Inclusos:

- PHP 7.4
- PostgreSql 11
- PgAdmin4
- Git
- Composer
- cURL
- Vim
- Redis
- Node e NPM
- PhpUnit
- Xdebug,
- Deployer

(Para mais detalhes consulte arquivo setup.sh)


Você vai precisar:
==============

- Virtualbox - https://www.virtualbox.org/
- Vagrant - http://www.vagrantup.com/
- Git - http://git-scm.com ( Opicional )
- Acesso Internet


-> Instale o Virtualbox e o Vagrant de acordo com seu sistema operacional. ( A instalação é bem simples e pode ser feita sem muitas dificuldades... )


Modo de Uso
===========

A instalação inicial pode ser feita de duas formas, sendo:

1º -> Caso tenha o Git instaldo em sua máquina, proceda da seguinte forma:



* Clone esse repositório para sua máquina:

- git clone https://github.com/philipsilv/vagrant-setup-php.git ( Execute o comando no seu Prompt/Terminal )

* Ainda no Prompt de comando/Terminal Entre no diretório vagrant-setup-php  ( Ou o que você definir na hora da clonagem )

* Inicie a máquina virtual com o comando:

- vagrant up




2º -> Caso não tenha o Git instalado em sua máquina, acesse https://github.com/philipsilv/vagrant-setup-php baixe o arquivo do projeto no formato .zip e descompacte-o onde desejar.

* Abra seu Prompt/Terminal , acesse o diretorio que acabou de descompactar e rode o comando:

- vagrant up




Após este comando 'vagrant up', o Vagrant ficará responsavel por baixar o sistema operacional ( neste caso Ubuntu Server 64 ), configurar a máquina virtual no VirtualBox e posteriormente baixar, instalar e configurar todos os pacotes do script 'setup.sh' (Sim! A primeira vez realmente é um pouco mais demorado).

Quando tudo estiver pronto, um servidor web estará disponível no endereço http://localhost:8080.


Clique [aqui](./php7-apache/phppgadmin.md) para configurar  o pgadmin. Após terminar a configuração acesse o endereço http://localhost:8080/phppgadmin. para acessar utilize:

- Login: vagrant
- Senha: vagrant

obs:(A senha padrão para todos os serviços é vagrant).


Coloque seu código no diretório "www". Todo o conteúdo dele estará disponível via http://localhost:8080. (Como teste, já existe um arquivo index.php que chama a função phpinfo() ).

Para desligar a máquina virtual utilize o comando:

- vagrant halt

Para religar novamente utilize:

- vagrant up

Caso queira destruir a máquina virtual (o conteúdo do www não será excluido):

- vagrant destroy


CHANGELOG.md
# Changelog

Todas as alterações significativas desse repositório serão documentadas neste arquivo.

## [Unreleased]

## [2.5.0] - 25/12/2019
### Adicionado
- Suporte ao php7.4 e postgresql server

### Alterações
- Atualização na documentação

### Descontinuado
- 

### Correções
- Nada

### Exclusões
- mysql

### Segurança
- Nada

-----------------------

## [2.4.0] - 20/08/2017
### Adicionado
- Suporte ao Deployer

### Alterações
- Atualização na documentação

### Descontinuado
- Nada

### Correções
- Nada

### Exclusões
- Nada

### Segurança
- Nada

-----------------------

## [2.3.0] - 20/08/2017
### Adicionado
- Suporte ao Node/NPM, PhpUnit e Xdebug

### Alterações
- Atualização na documentação

### Descontinuado
- Nada

### Correções
- Nada

### Exclusões
- Nada

### Segurança
- Nada

-----------------------

## [Unreleased]

## [2.0.0] - 20/08/2017
### Adicionado
- Suporte ao PHP 7.1

### Alterações
- Atualização na documentação

### Descontinuado
- Nada

### Correções
- Nada

### Exclusões
- Nada

### Segurança
- Nada

[Unreleased]: https://github.com/carlosfgti/vagrant-setup-php/compare/2.0.0...HEAD
[2.0.0]: https://github.com/carlosfgti/vagrant-setup-php/compare/v1.1.4...v2.0.0


php7-apache/web_setup.sh
#!/bin/bash

echo "---- Iniciando instalacao do ambiente de Desenvolvimento PHP ---"

echo "--- Atualizando lista de pacotes ---"
sudo apt-get update

echo "--- Instalando pacotes basicos ---"
sudo apt-get install software-properties-common vim curl python-software-properties git-core --assume-yes --force-yes

echo "--- Adicionando repositorio do pacote PHP ---"
sudo add-apt-repository ppa:ondrej/php

echo "--- Atualizando lista de pacotes ---"
sudo apt-get update

echo "--- Instalando PHP, Apache e alguns modulos ---"

sudo apt-get install php7.3 php7.3-common --assume-yes --force-yes
sudo apt-get install php7.3-cli libapache2-mod-php7.3 php7.3-mysql php7.3-pgsql php7.3-curl php-memcached php7.3-dev php7.3-mcrypt php7.3-sqlite3 php7.3-mbstring php*-mysql  php-gd php-xml php-mbstring  zip unzip --assume-yes --force-yes

echo "--- Habilitando o PHP 7 ---"
sudo a2dismod php5
sudo a2enmod php7.3

echo "--- Habilitando mod-rewrite do Apache ---"
sudo a2enmod rewrite

echo "--- Reiniciando Apache ---"
sudo service apache2 restart

echo "--- Baixando e Instalando Composer ---"
curl -sS https://getcomposer.org/installer | php
sudo mv composer.phar /usr/local/bin/composer

echo "--- Instalando Banco NoSQL -> Redis <- ---"
sudo apt-get install redis-server --assume-yes
sudo apt-get install php7.3-redis --assume-yes

echo "--- Baixando e Instalando o Node ---"
curl -sL https://deb.nodesource.com/setup_13.x | sudo -E bash -
sudo apt-get install -y nodejs --assume-yes --force-yes

echo "--- Baixando e Instalando o PhpUnit ---"
wget https://phar.phpunit.de/phpunit-8.0.phar
sudo chmod +x phpunit-8.0.phar --assume-yes --force-yes
sudo mv phpunit-8.0.phar /usr/local/bin/phpunit
phpunit --version

echo "--- Baixando e Instalando o deployer ---"
curl -LO https://deployer.org/deployer.phar
sudo mv deployer.phar /usr/local/bin/dep
sudo chmod +x /usr/local/bin/dep

echo "--- Baixando e Instalando o Xdebug ---"
git clone https://github.com/xdebug/xdebug.git /tmp/xdebug
cd /tmp/xdebug/
sudo ./rebuild.sh --assume-yes --force-yes

echo "[OK] --- Ambiente de desenvolvimento concluido ---"


setup.sh
#!/bin/bash

echo "---- Iniciando instalacao do ambiente de Desenvolvimento PHP [EspecializaTI] ---"

echo "--- Atualizando lista de pacotes ---"
sudo apt-get update

echo "--- Definindo Senha padrao para o MySQL e suas ferramentas ---"

DEFAULTPASS="vagrant"
sudo debconf-set-selections <<EOF
mysql-server	mysql-server/root_password password $DEFAULTPASS
mysql-server	mysql-server/root_password_again password $DEFAULTPASS
dbconfig-common	dbconfig-common/mysql/app-pass password $DEFAULTPASS
dbconfig-common	dbconfig-common/mysql/admin-pass password $DEFAULTPASS
dbconfig-common	dbconfig-common/password-confirm password $DEFAULTPASS
dbconfig-common	dbconfig-common/app-password-confirm password $DEFAULTPASS
phpmyadmin		phpmyadmin/reconfigure-webserver multiselect apache2
phpmyadmin		phpmyadmin/dbconfig-install boolean true
phpmyadmin      phpmyadmin/app-password-confirm password $DEFAULTPASS
phpmyadmin      phpmyadmin/mysql/admin-pass     password $DEFAULTPASS
phpmyadmin      phpmyadmin/password-confirm     password $DEFAULTPASS
phpmyadmin      phpmyadmin/setup-password       password $DEFAULTPASS
phpmyadmin      phpmyadmin/mysql/app-pass       password $DEFAULTPASS
EOF

echo "--- Instalando pacotes basicos ---"
sudo apt-get install software-properties-common vim curl python-software-properties git-core --assume-yes --force-yes

echo "--- Adicionando repositorio do pacote PHP ---"
sudo add-apt-repository ppa:ondrej/php

echo "--- Atualizando lista de pacotes ---"
sudo apt-get update

echo "--- Instalando MySQL, Phpmyadmin e alguns outros modulos ---"
sudo apt-get install mysql-server-5.5 mysql-client phpmyadmin --assume-yes --force-yes

echo "--- Instalando PHP, Apache e alguns modulos ---"
sudo apt-get install php7.1 php7.1-common --assume-yes --force-yes
sudo apt-get install php7.1-cli libapache2-mod-php7.1 php7.1-mysql php7.1-curl php-memcached php7.1-dev php7.1-mcrypt php7.1-sqlite3 php7.1-mbstring php*-mysql  php-gd php-xml php-mbstring  zip unzip --assume-yes --force-yes

echo "--- Habilitando o PHP 7.1 ---"
sudo a2dismod php5
sudo a2enmod php7.1

echo "--- Habilitando mod-rewrite do Apache ---"
sudo a2enmod rewrite

echo "--- Reiniciando Apache ---"
sudo service apache2 restart

echo "--- Baixando e Instalando Composer ---"
curl -sS https://getcomposer.org/installer | php
sudo mv composer.phar /usr/local/bin/composer

echo "--- Instalando Banco NoSQL -> Redis <- ---"
sudo apt-get install redis-server --assume-yes
sudo apt-get install php7.1-redis --assume-yes

# Instale apartir daqui o que você desejar

echo "--- Baixando e Instalando o Node ---"
curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
sudo apt-get install -y nodejs --assume-yes --force-yes

echo "--- Baixando e Instalando o PhpUnit ---"
wget https://phar.phpunit.de/phpunit-8.0.phar
sudo chmod +x phpunit-8.0.phar --assume-yes --force-yes
sudo mv phpunit-8.0.phar /usr/local/bin/phpunit
phpunit --version

echo "--- Baixando e Instalando o deployer ---"
curl -LO https://deployer.org/deployer.phar
sudo mv deployer.phar /usr/local/bin/dep
sudo chmod +x /usr/local/bin/dep

echo "--- Baixando e Instalando o Xdebug ---"
git clone https://github.com/xdebug/xdebug.git /tmp/xdebug
cd /tmp/xdebug/
sudo ./rebuild.sh --assume-yes --force-yes
echo 'Adicione no arquivo php.ini a seguinte linha: zend_extension="xdebug.so"'

echo "[OK] --- Ambiente de desenvolvimento concluido ---"
